### PR TITLE
Use maven to publish to local repository

### PR DESCRIPTION
**Pull request by AmauCgy**
The publish-local copied directly the files into the maven repository, causing issues when working on SNAPSHOT versions. For example, maven always downloaded from the remote repository because it didn't find the local version to be newer.
We now use a maven command which update every information that has to be.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@lyra-network/openapi-dev-tool",
-  "version": "9.7.0",
+  "version": "9.8.0",
```
Comment (gigaga) [inline package.json:3]: Please use semver semantic. Indeed, if the goal of your PR is to fix an issue, you have to use 9.7.1

File: src/commands/publish-local.js [hunk 2]
```diff
@@ -78,29 +77,25 @@ export function publishLocal(config = { config: { specs: [] } }) {
...
             );
 
-            fs.writeFileSync(`${target}/${artifactName}.pom`, pomContent);
+            const pomFile = `${getTempDir().name}/${
+              path.parse(archive).name
+            }.pom`;
+            fs.writeFileSync(pomFile, pomContent);
+
+            // Install with maven plugin
+            const mvn = require('maven').create({ quiet: true });
```
Comment (gigaga) [inline src/commands/publish-local.js:94]: Please declare this module at the beginning of the JS file

File: src/commands/publish-local.js [hunk 3]
```diff
@@ -78,29 +77,25 @@ export function publishLocal(config = { config: { specs: [] } }) {
...
+            await mvn.execute(['install:install-file'], {
```
Comment (gigaga) [inline src/commands/publish-local.js:95]: Please, move this mvn operation into the dedicated maven module (`./lib/maven.js`)

Comment (gigaga) [inline src/commands/publish-local.js:95]: With your code, is seems that `--repoPath` parameter is no longer used. You should change README, config-validation and config-definition files

File: src/commands/publish-local.js [hunk 4]
```diff
@@ -2,11 +2,10 @@ import colors from 'colors';
 import path from 'path';
 import { paramCase } from 'change-case';
 import settle from 'promise-settle';
-import mkdirp from 'mkdirp';
 import fs from 'fs';
 
 import { generateSpecsArchive } from '../lib/archiver';
-import { getTempDir, getPOMContent } from '../lib/utils';
+import { getPOMContent, getTempDir } from '../lib/utils';
```
Comment (gigaga) [inline src/commands/publish-local.js:8]: Why this change?

File: src/commands/publish-local.js [hunk 5]
```diff
@@ -43,7 +42,7 @@ export function publishLocal(config = { config: { specs: [] } }) {
             archive = await generateSpecsArchive(api, fileToArchive);
 
             // Find doublon
-            if (artifactIds.indexOf(paramCase(api.info.title)) != -1) {
+            if (artifactIds.indexOf(paramCase(api.info.title)) !== -1) {
```
Comment (gigaga) [inline src/commands/publish-local.js:45]: 👍

**Review by gigaga — CHANGES_REQUESTED** [review]

File: src/commands/publish-local.js (hunk 4 above)
Comment (AmauCgy) [reply src/commands/publish-local.js:8]: Auto formatting i guess, no real necessity

File: src/lib/maven.js [hunk 6]
```diff
@@ -6,6 +6,8 @@ import commandExists from 'command-exists';
 import AdmZip from 'adm-zip';
 import { downloadFile } from './utils';
 
+import maven from 'maven';
```
Comment (gigaga) [inline src/lib/maven.js:9]: Please start your modules imports by the "public" modules, then yours

File: src/lib/maven.js [hunk 7]
```diff
@@ -79,3 +81,11 @@ export function downloadArtifact(artifact, urlDownloadTemplate, verbose) {
       });
   });
 }
+
+export async function installToLocalRepository(archive, pomFile) {
```
Comment (gigaga) [inline src/lib/maven.js:85]: It is good, however, by removing `repoPath`, we are no longer able to install file without having `mvn` command :(.
However, I'm not sure if that was really usable.

**Review by gigaga — CHANGES_REQUESTED** [review]
